### PR TITLE
Add options `ignoreInPageNav` and `ignoreInSearch` to headings

### PR DESCRIPTION
**Pull request by colinrotherham**
This is a quick supporting PR for:

* https://github.com/alphagov/govuk-design-system/pull/3106

I've extended page heading `aliases` to also include `url`, `ignoreInPageNav` and `ignoreInSearch`

### Before
Heading aliases were configurable

```yaml
headingAliases:
  Section break: horizontal rule, hr
```

### After
Headings can now be excluded from the side navigation, search, or given a custom URL fragment

```yaml
headings:
  - text: Section break
    aliases: horizontal rule, hr

  - text: Section example
    url: section-override

  - text: Section link not indexed or shown
    ignoreInPageNav: true

  - text: Section link not indexed only
    ignoreInSearch: true
```

### CODE REVIEW

**Comment by frankieroberto** [comment]
@colinrotherham thanks - this is much cleaner than resorting to HTML! 😄 

My only comment is that it might not be clear that `ignoreInPageNav` also affects the site search? Maybe that's implied though? Or alternatively is it worth having a separate `ignoreInSiteSearch` in case you'd ever want something to appear in the side nav but not in the site search (or vice versa)?

**Review by frankieroberto — APPROVED** [review]
👍

**Review by domoscargin — APPROVED** [review]
Nice, looks good.

Looks like the only diff in compiled code is to the search index, which is to be expected - some added "aliases" properties here and there.

File: config/navigation.js [hunk 1]
```diff
@@ -6,36 +6,31 @@
 module.exports = [
   {
     label: 'Get started',
-    url: 'get-started',
-    includeInSearch: true
+    url: 'get-started'
```
Comment (domoscargin) [inline config/navigation.js:9]: 👍

**Pull request merged**